Assets/MixedRealityToolkit.Providers/XRSDK/GenericXRSDKSpatialMeshObserver.cs

```diff
@@ -0,0 +1,482 @@
+            string name = null,
+            uint priority = DefaultPriority,
+            BaseMixedRealityProfile profile = null) : base(spatialAwarenessSystem, name, priority, profile)
+        { }
+
+        #region BaseSpatialObserver Implementation
+
+        /// <summary>
+        /// Creates the XRMeshSubsystem and handles the desired startup behavior.
+        /// </summary>
+        protected override void CreateObserver()
+        {
+            if (SpatialAwarenessSystem == null) { return; }
+
+            if (XRSDKSubsystemHelpers.MeshSubsystem != null)
+            {
+                ConfigureObserverVolume();
+
+                if (StartupBehavior == AutoStartBehavior.AutoStart)
+                {
+                    Resume();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Implements proper cleanup of the SurfaceObserver.
+        /// </summary>
+        protected override void CleanupObserver()
+        {
+            if (IsRunning)
+            {
+                Suspend();
+            }
+
+            // Since we don't handle the mesh subsystem's lifecycle, we don't do anything more here.
+        }
+
+        #endregion BaseSpatialObserver Implementation
+
+        #region BaseSpatialMeshObserver Implementation
+
+        /// <inheritdoc />
+        protected override int LookupTriangleDensity(SpatialAwarenessMeshLevelOfDetail levelOfDetail)
+        {
+            // For non-custom levels, the enum value is the appropriate triangles per cubic meter.
+            int level = (int)levelOfDetail;
+            if (XRSDKSubsystemHelpers.MeshSubsystem != null)
+            {
+                XRSDKSubsystemHelpers.MeshSubsystem.meshDensity = level / (float)SpatialAwarenessMeshLevelOfDetail.Fine; // For now, map Coarse to 0.0 and Fine to 1.0
+            }
+            return level;
+        }
+
+        #endregion BaseSpatialMeshObserver Implementation
+
+        #region IMixedRealityCapabilityCheck Implementation
+
+        /// <inheritdoc />
+        public bool CheckCapability(MixedRealityCapability capability)
+        {
+            if (capability != MixedRealityCapability.SpatialAwarenessMesh)
+            {
+                return false;
+            }
+
+            var descriptors = new List<XRMeshSubsystemDescriptor>();
+            SubsystemManager.GetSubsystemDescriptors(descriptors);
+
+            return descriptors.Count > 0;
+        }
+
+        #endregion IMixedRealityCapabilityCheck Implementation
+
+        #region IMixedRealityDataProvider Implementation
+
+        /// <inheritdoc />
+        public override void Update()
+        {
+            UpdateObserver();
+        }
+
+        #endregion IMixedRealityDataProvider Implementation
+
+        #region IMixedRealitySpatialAwarenessObserver Implementation
+
+        /// <summary>
+        /// A queue of TrackableId that need their meshes created (or updated).
```

Comment:
```suggestion
        /// A queue of MeshId that need their meshes created (or updated).
```

Reply:
Or maybe there's some subtyping here that isn't immediately obvious

Reply:
Type name updated between an early XR SDK revision and now, but I didn't update the comment :) thanks!